From c2a736a767cbe72b03f2ffec47b68f77cb65fc6d Mon Sep 17 00:00:00 2001
From: zhang sanshan <b51434@freescale.com>
Date: Thu, 8 Jan 2015 10:47:09 +0800
Subject: [PATCH 19/34] MA-6096 add '-s' for mkfs.ubifs

the option '-s' define file_contexts for /system when we enable selinux.

Signed-off-by: zhang sanshan <b51434@freescale.com>
---
 tools/releasetools/build_image.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/releasetools/build_image.py b/tools/releasetools/build_image.py
index b63dcd6..1ee9483 100755
--- a/tools/releasetools/build_image.py
+++ b/tools/releasetools/build_image.py
@@ -303,6 +303,9 @@ def BuildImage(in_dir, prop_dict, out_file, target_out=None):
     build_command.append(in_dir)
     build_command.append("-o")
     build_command.append(out_file)
+    build_command.append("-s")
+    if "selinux_fc" in prop_dict:
+      build_command.append(prop_dict["selinux_fc"])
     if prop_dict.get("mkfsubifs_flags", None):
       build_command.extend(prop_dict["mkfsubifs_flags"].split())
   else:
@@ -430,7 +433,6 @@ def ImagePropFromGlobalDict(glob_dict, mount_point):
       )
   for p in common_props:
     copy_prop(p, p)
-
   copy_prop("mkfsubifs_flags", "mkfsubifs_flags")
   d["mount_point"] = mount_point
   if mount_point == "system":
-- 
1.8.0

